### ISSUE TITLE
`--allow-local-files` fails on chromium in snap environment

### ISSUE THREAD

**Issue by matoken** [marp-team/marp-cli#201]
It happens in npm, npx, Standalone binary in Debian sid adm64 environment.

```
$ marp --allow-local-files --pdf test.md
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from test.md.
[  WARN ] Marp CLI has detected accessing to local file. That is blocked by security reason. Instead we recommend using assets uploaded to
          online. (Or you can use --allow-local-files option if you are understood of security risk)
[ ERROR ] Failed converting Markdown. (net::ERR_FILE_NOT_FOUND at file:///tmp/tmp-1187272jpnBmJk9frtF.html)
```

Chromium in this environment includes a snap version and a Debian package version.

```
$ which chromium
/snap/bin/chromium
$ PATH=`echo $PATH|tr : '\n'|grep -v snap|tr "\n" :` which chromium
/usr/bin/chromium
```

Using the package version instead of the snap version worked fine.

```
$ PATH=`echo $PATH|tr : '\n'|grep -v snap|tr "\n" :` marp --allow-local-files --pdf hoge.md
```

**Comment by yhatt** [comment]
See also similar issue happend by Snap Chromium: https://github.com/marp-team/marp-vscode/issues/94

**Comment by matoken** [comment]
Thank you.
Chromium could be specified by using `CHROME_PATH`.
It seems to be helpful if a warning is displayed when using the snap version of Chromium.

**Comment by yhatt** [comment]
The suggested warning may be helpful for CLI user, but does not resolve marp-team/marp-vscode#94 so thinking about radical solution would be better.

Owing to your report, we recognized the Snapd Chromium cannot access from the sandbox container to user-land `/tmp/`, due to [strict snap confinement](https://snapcraft.io/docs/snap-confinement). Snapd Chromium only allows [reading visible files from `$HOME`](https://snapcraft.io/docs/home-interface).

Marp CLI should create visible tmp file into `$HOME` when the resolved Chromium path starts with `/snap`.

**Comment by yhatt** [comment]
Fixed in [v0.17.1](https://github.com/marp-team/marp-cli/releases/tag/v0.17.1).

**Comment by zhelnio** [comment]
Looks like this bug still appears
- Ubuntu 20.04.1 LTS, updated
- chromium installed using the system default **apt install** (installs snap package)

```
$ npx marp slides_ru.md --allow-local-files --output out/slides_ru.pdf
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from slides_ru.md.
[  WARN ] The local file is missing and will be ignored. Make sure the file path is correct.
[ ERROR ] Failed converting Markdown. (net::ERR_FILE_NOT_FOUND at file:///tmp/tmp-32659-D1CkptSeH8No-.html)

$ which chromium
/snap/bin/chromium

$ npx marp --version
@marp-team/marp-cli v0.21.0 (w/ @marp-team/marp-core v1.3.0)

```

**Comment by yhatt** [comment]
@zhelnio Thanks!

Previously Marp CLI could find out `/snap/bin/chromium` and [detect the snap chromium from the path starting with `/snap/`](https://github.com/marp-team/marp-cli/pull/203/files#diff-d1aec63891ce155808ea60d45b8a2ba7R361-R365). But now, Marp is trying to use `/usr/bin/chromium-browser`, that is a proxy script for calling `/snap/bin/chromium`.

```bash
yhatt@yhatt-ubuntu2004:~$ echo '# Marp' | DEBUG=puppeteer:launcher npx @marp-team/marp-cli -o ~/marp.pdf --allow-local-files
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from -.
  puppeteer:launcher Calling /usr/bin/chromium-browser --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=TranslateUI --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --user-data-dir=/tmp/marp-cli-conversion --headless --hide-scrollbars --mute-audio about:blank --export-tagged-pdf --remote-debugging-pipe +0ms
[  WARN ] The local file is missing and will be ignored. Make sure the file path is correct.
[ ERROR ] Failed converting Markdown. (net::ERR_FILE_NOT_FOUND at file:///tmp/tmp-9964-KRtSNXzO5F7m-.html)

yhatt@yhatt-ubuntu2004:~$ tail /usr/bin/chromium-browser 
      }
    }
EOF
  )"
  qdbus org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript "$SCRIPT" 2> /dev/null
fi

# TODO: handle other desktop environments

exec /snap/bin/chromium "$@"
```

Probably we should check the path of "real" process launched by Puppeteer.

**Comment by zhelnio** [comment]
@yhatt I've tried to set CHROME_PATH but is does not help
```
$ CHROME_PATH=/snap/bin/chromium npx marp slides_ru.md --allow-local-files --output out/slides_ru.pdf
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from slides_ru.md.
[ ERROR ] Failed converting Markdown. (dir option must be relative to "/tmp", found "/home/user".)
```

**Comment by yhatt** [comment]
I think it may be another problem that has brought from #264 (released in v0.20.0). Is `CHROME_PATH=/snap/bin/chromium npx @marp-team/marp-cli@0.19.0` working?

**Comment by zhelnio** [comment]
No, it does not
```
$ CHROME_PATH=/snap/bin/chromium npx @marp-team/marp-cli@0.19.0 slides_ru.md --allow-local-files --output out/slides_ru.pdf
npx: installed 271 in 24.933s
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from slides_ru.md.
[ ERROR ] Failed converting Markdown. (dir option must be relative to "/tmp", found "/home/user".)
```

**Comment by yhatt** [comment]
Ah, it didn't really a real matter. Probably we've missed [a breaking change of node-tmp to create tmpfile for conversion](https://github.com/raszi/node-tmp/blob/v0.2.0/CHANGELOG.md).

**Comment by yhatt** [comment]
Fixed again in [v0.21.1](https://github.com/marp-team/marp-cli/releases/tag/v0.21.1).

**Comment by nevyn** [comment]
I'm getting this issue in marp-cli 2.1.2 :(

```
❯ chromium --version
Chromium 104.0.5112.101 snap

❯ npx @marp-team/marp-cli@latest --allow-local-files 04-lektion.md -o renders/04-lektion.pdf
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from 04-lektion.md.
[  WARN ] Some of the local files are missing and will be ignored. Make sure the file paths are correct.
[  INFO ] 04-lektion.md => renders/04-lektion.pdf
```

... and all images are missing. exporting from vscode isn't working either.

**Comment by nevyn** [comment]
I'm getting this again :'(

```bash
nevyn@itinf-nevyn:~/Dev/nackademin-linux2$ CHROME_PATH=/snap/chromium/current/usr/lib/chromium-browser/chrome npx @marp-team/marp-cli@2.1.4 --allow-local-files preprocessed/01-lektion.md -o renders/01-lektion.pdf
[  INFO ] Converting 1 markdown...
[  WARN ] Insecure local file accessing is enabled for conversion from
          preprocessed/01-lektion.md.
[  WARN ] Some of the local files are missing and will be ignored. Make sure the
          file paths are correct.
[  INFO ] preprocessed/01-lektion.md => renders/01-lektion.pdf
nevyn@itinf-nevyn:~/Dev/nackademin-linux2$ cat preprocessed/01-lektion.md | grep '\!\['
![bg](img/companies.png)
![bg](img/unix-family.jpg)
![bg](img/gnu-linux-distribution-timeline.png)
![](img/passwd-file.png)
![bg right contain drop-shadow](img/pki.png)
nevyn@itinf-nevyn:~/Dev/nackademin-linux2$ ls img/companies.png img/unix-family.jpg img/gnu-linux-distribution-timeline.png img/passwd-file.png img/pki.png
img/companies.png                        img/pki.png
img/gnu-linux-distribution-timeline.png  img/unix-family.jpg
img/passwd-file.png
```

so as you can see, all the referenced files are indeed available...

... UGH and now I'm realizing that the image paths are of course relative to the markdown file, not relative to cwd. adding a symbolic link to img inside preprocessed fixed it for me. 

Another issue I ran into is I accidentally used some absolute paths, like this: `![](/img/foo.png)`. Removing the initial `/` fixed it of course.

Leaving these here so Future Googlers see my solutions.
